static/js/settings_users.js

```diff
@@ -601,6 +601,45 @@ function handle_human_form(tbody, status_field) {
         const modal = ret.modal;
         const fields_user_pills = ret.fields_user_pills;
 
+        const stream_ids = stream_data.get_default_stream_ids();
+        const default_streams = [];
+
+        for (const stream_id of stream_ids) {
+            const stream_name = stream_data.maybe_get_stream_name(stream_id);
+            default_streams.push({name: stream_name});
+        }
+
+        modal.find("#subscribe-user-to-default-streams").on("click", (e) => {
+            e.preventDefault();
+            e.stopPropagation();
+
+            const url = "/json/users/me/subscriptions";
+            const data = {
+                subscriptions: JSON.stringify(default_streams),
+                principals: JSON.stringify([user_id]),
+            };
+
+            const spinner = $("#subscribe-user-spinner").expectOne();
+            spinner.fadeTo(0, 1);
+            loading.make_indicator(spinner, {text: i18n.t("Subscribing user")});
+            const remove_after = 1000;
+            const appear_after = 500;
+
+            channel.post({
+                url,
+                data,
+                success() {
+                    setTimeout(() => {
+                        ui_report.success(i18n.t("Subscribed"), spinner, remove_after);
+                        settings_ui.display_checkmark(spinner);
+                    }, appear_after);
```

Comment:
I think `ui_report.success` has an option to do this without a manual setTimeout?

Reply:
setTimeout in `ui_report.success` is for removing success message after it has been displayed; here we are using it for delaying success message.